### PR TITLE
Serve index.html from "/" and other fixes.

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

File: repo/github.go [hunk 1]
```diff
@@ -438,6 +436,12 @@ func CompleteDependencyRepo(name string, version string) *Repo {
 			Name:    name,
 			Version: version,
 			Domain:  "github.com"}
+	case "promise-polyfill":
+		return &Repo{
+			User:    "polymerlabs",
+			Name:    name,
+			Version: version,
+			Domain:  "github.com"}
 	case "web-compnent-tester":
```
Comment (cdata) [inline repo/github.go]: Is there a typo here? `compnent` -> `component`

File: gitproxy.go [hunk 2]
```diff
@@ -120,6 +120,9 @@ func proxyGitPathRequest(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if strings.HasSuffix(parsedReq.FilePath, "/") {
+		parsedReq.FilePath = parsedReq.FilePath + "index.html"
```
Comment (justinfagnani) [inline gitproxy.go]: Are there known cases where this is needed? I know serving index.html is _nearly_ ubiquitous, but it's not guaranteed, and so no one's reusable components should rely on this behavior. If they want index.html, they should say index.html.

File: repo/github.go [hunk 3]
```diff
@@ -500,10 +504,16 @@ func readDepsFromBower(contents []byte) (map[string]*Repo, error) {
 		}
 		org := orgSuffix[0]
 		repoVersion := strings.SplitN(orgSuffix[1], "#", 2)
+		var version string
```
Comment (justinfagnani) [inline repo/github.go]: Interesting that Go doesn't have a ternary operator. Seems like initializing the variable to the default might be idiomatic:

```
version := '*'
if len(repoVersion) == 2 {
    version = repoVersion[1]
}
```

wdyt?

File: gitproxy.go (hunk 2 above)
Comment (garlicnation) [inline gitproxy.go]: I wrote this to serve SPA demos. I'm not 100% sure that it's needed, but it makes it possible to share a demo that uses client-side routing and doesn't expect to see index.html in the URL.

**Comment by justinfagnani** [comment]
LGTM after addressing comments.

I don't know what to think about the index.html part. I'd like to encourage components that are written to work under web server configurations that don't serve index.html, like those that serve a directory listing (I want to do that for polyserve to make browsing served files easier).

File: repo/github.go (hunk 3 above)
Comment (garlicnation) [inline repo/github.go]: Way nicer looking.

File: gitproxy.go (hunk 2 above)
Comment (justinfagnani) [inline gitproxy.go]: true, it's going to be way more common to see this relied on at the app level. hmm...

**Pull request merged**